addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/SourceCodeDisclosureScanRule.java

```diff
@@ -688,6 +694,18 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
         }
     }
 
+    private static boolean isFontRequest(HttpMessage msg) {
+        try {
+            String path = msg.getRequestHeader().getURI().getPath();
+            if (path != null) {
+                return PATTERN_FONT_EXTENSIONS.matcher(path).find();
+            }
+        } catch (URIException e) {
+            log.error(e.getMessage(), e);
+        }
+        return false;
+    }
+
```

Comment:
This is the first duplication of this method.

Would it be preferable to have this as part of commonlib? Maybe something like http/HttpUtils or http/HttpMessageUtils. It could even be `isFontMessage(msg)` and include the `msg.getResponseHeader().hasContentType("font")` condition as well.

(For that matter should I be doing something similar for JavaScript and CSS? Ex: Migrate to commonlib with a single method.)

Reply:
I'm ok either way :)

Reply:
Sounds good to me, it could be even be something like `isResource(HttpMessage)` and include all 3 types (or others as needed, e.g. images).